src/tree.rs

```diff
@@ -76,15 +78,30 @@ impl Anchor {
 #[derive(Debug)]
 pub struct MerklePath {
     position: u32,
-    auth_path: [pallas::Base; MERKLE_DEPTH_ORCHARD],
+    auth_path: [MerkleHashOrchard; MERKLE_DEPTH_ORCHARD],
 }
 
 impl MerklePath {
     /// Generates a dummy Merkle path for use in dummy spent notes.
     pub(crate) fn dummy(mut rng: &mut impl RngCore) -> Self {
         MerklePath {
             position: rng.next_u32(),
-            auth_path: gen_const_array(|_| pallas::Base::random(&mut rng)),
+            auth_path: gen_const_array_with_default(MerkleHashOrchard::empty_leaf(), |_| {
+                MerkleHashOrchard(pallas::Base::random(&mut rng))
+            }),
+        }
+    }
+
+    /// Instantiates a new Merkle path given a leaf position and authentication path.
+    pub(crate) fn new(position: u32, auth_path: [pallas::Base; MERKLE_DEPTH_ORCHARD]) -> Self {
+        Self {
+            position,
+            auth_path: auth_path
```

Comment:
Can use `array::map` in future.

Reply:
Same question here; the path has a different value at each index.